### PR TITLE
[MM-20481] Migrate 'components/post_view/post_body_additional_content' module and associated tests to TypeScript

### PR DESCRIPTION
**Pull request by KuSh**
#### Summary

Migrate 'components/post_view/post_body_additional_content' module and associated tests to TypeScript

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/15460
JIRA: https://mattermost.atlassian.net/browse/MM-20481

### CODE REVIEW

**Comment by KuSh** [comment]
Hi, it's because I'm waiting for the PR I'm dependant on (https://github.com/mattermost/mattermost-webapp/pull/6669) to be merged before continuing on this 👍

**Comment by jasonblais** [comment]
@KuSh thanks for the heads up! I added a `Lifecycle/frozen` label so that it doesn't trigger the 'inactive PR' message :)

**Comment by KuSh** [comment]
> This PR has been automatically labelled "stale" because it hasn't had recent activity.
> A core team member will check in on the status of the PR to help with questions.
> Thank you for your contribution!
> 
> /cc @jasonblais @jfrerich @emilyacook

Unfortunately it depends (again) on an other PR (https://github.com/mattermost/mattermost-redux/pull/1272) to be merged :/

**Comment by devinbinnie** [comment]
> > This PR has been automatically labelled "stale" because it hasn't had recent activity.
> > A core team member will check in on the status of the PR to help with questions.
> > Thank you for your contribution!
> > /cc @jasonblais @jfrerich @emilyacook
> 
> Unfortunately it depends (again) on an other PR ([mattermost/mattermost-redux#1272](https://github.com/mattermost/mattermost-redux/pull/1272)) to be merged :/

@KuSh Sorry for the delay, I've merged the above redux PR. Feel free to update your branch and continue with your work!

**Comment by KuSh** [comment]
> @KuSh Sorry for the delay, I've merged the above redux PR. Feel free to update your branch and continue with your work!
No problem, it wasn't a criticism, just an explanation for the absence of activity. Anyway, I've pushed a new version :+1:

File: components/post_view/post_attachment_opengraph/post_attachment_opengraph.tsx [hunk 1]
```diff
@@ -141,7 +141,7 @@ export default class PostAttachmentOpenGraph extends React.PureComponent<Props>
             props,
         };
 
-        return this.props.actions.editPost(patchedPost);
+        return this.props.actions?.editPost(patchedPost);
```
Comment (devinbinnie) [inline components/post_view/post_attachment_opengraph/post_attachment_opengraph.tsx:144]: `actions` should never be nullable in a redux-connected component.
What's likely happening is by importing `Props` from this component in the redux connector, the TS compiler is expecting all uses of the component to also pass in `actions` as a property to the redux-connected component.

You'll want to define a type in the redux connector called `OwnProps` that will only contain the props that the redux connector needs to run `mapStateToProps`.

File: package-lock.json [hunk 2]
```diff
@@ -13484,9 +13484,9 @@
       }
     },
     "icu4c-data": {
-      "version": "0.63.2",
-      "resolved": "https://registry.npmjs.org/icu4c-data/-/icu4c-data-0.63.2.tgz",
-      "integrity": "sha512-vT6/47CcBzDemlhRzkL7dZLoNvuUWjjiuKZHMt5T4dbkKAqLBh7ZQ33GU13ecC/aIcMlIdBOqtF4uIYTgWML4Q==",
+      "version": "0.64.2",
```
Comment (devinbinnie) [inline package-lock.json:13487]: Can we revert these changes and just keep the changes to `mattermost-redux`?

**Review by devinbinnie — COMMENTED** [review]
Thanks for the PR @KuSh!
I've added a couple comments for you to review.

File: package-lock.json (hunk 2 above)
Comment (KuSh) [reply package-lock.json:13487]: Sure. I did let it because it was automatically added by npm while installing the new mattermost-redux version but sure I'll remove it !

File: components/post_view/post_attachment_opengraph/post_attachment_opengraph.tsx (hunk 1 above)
Comment (KuSh) [reply components/post_view/post_attachment_opengraph/post_attachment_opengraph.tsx:144]: I'll see what I can do, thanks

Comment (KuSh) [reply components/post_view/post_attachment_opengraph/post_attachment_opengraph.tsx]: I followed your advice, thanks

**Review by devinbinnie — APPROVED** [review]
LGTM! Thanks @KuSh!

**Comment by KuSh** [comment]
> LGTM! Thanks @KuSh!

Thanks ! It has been a pretty long journey ^^

**Review by hmhealey — APPROVED** [review]

File: components/post_view/post_body_additional_content/post_body_additional_content.tsx [hunk 3]
```diff
@@ -111,7 +93,6 @@ export default class PostBodyAdditionalContent extends React.PureComponent {
 
                 return (
                     <YoutubeVideo
-                        postId={this.props.post.id}
```
Comment (saturninoabril) [inline components/post_view/post_body_additional_content/post_body_additional_content.tsx:114]: I got error after posting a Youtube link (ex. "https://www.youtube.com/watch?v=a9__D53WsUs"). Might be related to above.
Screen Shot 2020-11-11 at 12 58 32 AM

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @KuSh! I did some tests and looked good except it crashed after posting a youtube link -- see comment. Please take a look.

File: components/post_view/post_body_additional_content/post_body_additional_content.tsx (hunk 3 above)
Comment (hmhealey) [reply components/post_view/post_body_additional_content/post_body_additional_content.tsx:114]: Oh yeah, that's required by the YoutubeVideo's mapStateToProps, but it's not required by the component itself. That'll be resolved properly when that component is migrated to TS, but for now, you can just add it `postId` to the YoutubeVideo's propTypes as required.

**Comment by saturninoabril** [comment]
@KuSh Please ping whenever ready for another review.  Thanks!

File: components/post_view/post_body_additional_content/post_body_additional_content.tsx (hunk 3 above)
Comment (KuSh) [reply components/post_view/post_body_additional_content/post_body_additional_content.tsx]: Thanks for the test and sorry for the bug (and the delay), I did push a fix :+1:

Comment (KuSh) [reply components/post_view/post_body_additional_content/post_body_additional_content.tsx]: Tests are failing I'll look at it

Comment (KuSh) [reply components/post_view/post_body_additional_content/post_body_additional_content.tsx]: That should be ok now

**Comment by KuSh** [comment]
@saturninoabril Could you take another bit of your time to review my PR ?

**Review by saturninoabril — APPROVED** [review]
Thanks @KuSh, LGTM!
